framework-cloud/framework-maven/src/main/java/org/kie/cloud/maven/MavenDeployer.java

```diff
@@ -52,33 +54,13 @@ public static void buildAndDeployMavenProject(String basedir) {
      * @param buildCommand Build command, for example "install" or "deploy".
      */
     private static void buildMavenProject(String basedir, String buildCommand) {
-        String buildSettings = System.getProperty("kjars.build.settings.xml");
-
-        // need to backup (and later restore) the current class loader, because the Maven/Plexus does some classloader
-        // magic which then results in CNFE in RestEasy client
-        // run the Maven build which will create the kjar. The kjar is then either installed or deployed to local and
-        // remote repo
-        logger.debug("Building and deploying Maven project from basedir '{}'.", basedir);
-        ClassLoader classLoaderBak = Thread.currentThread().getContextClassLoader();
-        System.setProperty("maven.multiModuleProjectDirectory", basedir); // required by MavenCli 3.3.0+
-
-        List<String> mvnArgs = new ArrayList<String>(Arrays.asList("-B", "-e", "clean", buildCommand));;
-
-        // use custom settings.xml file, if one specified
-        if (buildSettings != null && !buildSettings.isEmpty()) {
-            mvnArgs.add("-s");
-            mvnArgs.add(buildSettings);
-        }
-
-        MavenCli cli = new MavenCli();
-        int mvnRunResult = cli.doMain(mvnArgs.toArray(new String[mvnArgs.size()]), basedir, System.out, System.err);
-
-        Thread.currentThread().setContextClassLoader(classLoaderBak);
+        try {
+            MavenUtil mavenUtil = MavenUtil.forProject(Paths.get(basedir), Paths.get(SETTINGS_XML_PATH)).forkJvm();
```

Comment:
Will this work if someone forgets to set `kjars.build.settings.xml` system property? (maybe it's handled already in the MavenUtil. If not, perhaps check that that file exists before calling MavenUtil..

Reply:
Added handling of settings.xml file. It doesn't need to be defined (in case when user just want to install project or deploy project to the repo without authentication), but if it is defined then it has to point to existing location.